### ISSUE TITLE
Empty title ID declaration produces no ID

### ISSUE THREAD

**Issue by ghost** [fletcher/MultiMarkdown#19]
This could very well be an intentional feature, so feel free to dismiss it. I noticed that if a title is declared with an empty ID declaration, the h element acquires no ID at all. To me it would make more sense if the original ID generating mechanism produced an ID. Example:

```
 # Chapter Title [] #
```

Currently produces:

```
 <h1>Chapter Title</h1>
```

But could produce:

```
 <h1 id="chaptertitle">Chapter Title</h1>
```

Where the current behaviour creates a conflict is in the xhtml-toc.xslt. Any titles with empty ID declarations will result in duplicate entries, breaking the ToC linkages.

**Comment by fletcher** [comment]
Hmmm...  It's not an intentional feature, but I'm not 100% certain it's a bug.

Under what circumstances would you want to use an empty bracket set in the title?

**Comment by ghost** [comment]
It's a bit of an edge case, admittedly, and not so much a desire to use an empty bracket, as a side effect. If I had a program which could hypothetically insert custom values into the title field, but this hypothetical program (ahem) did not have a logical language built in to its export engine, and if sections of the export did not have the custom value set, it would result in an empty value.

So for example two documents:
- Document A has a custom meta-data value of "prefs_general" for DocumentID.
- Document B has no value set for DocumentID

The export engine has been instructed to suffix each title with:

```
[<$custom:DocumentID>]
```

Which produces interpolated results of:

```
# Document A [prefs_general] #

# Document B [] #
```

In short, it's a way of keeping custom IDs out of the way in the program, reducing clutter in its internal display of the documents.

**Comment by fletcher** [comment]
For now, I think I'm going to leave this as a "feature" instead of a bug.
